hummingbot/connector/exchange/bittrex/bittrex_exchange.py

```diff
@@ -0,0 +1,403 @@
+                exchange_order_id=order_id,
+            )
+            self._order_tracker.process_order_update(order_update=order_update)
+
+    async def _process_execution_event(self, events: Dict[str, Any]):
+        for execution_event in events:
+            order_id = execution_event["orderId"]
+            tracked_order: InFlightOrder = None
+            for order in self._order_tracker.all_fillable_orders.values():
+                if order.exchange_order_id is None:
+                    continue
+                exchange_order_id = await order.get_exchange_order_id()
+                if exchange_order_id == order_id:
+                    tracked_order = order
+                    break
+            if tracked_order is not None:
+                percent_token = tracked_order.quote_asset
+                fee = TradeFeeBase.new_spot_fee(
+                    fee_schema=self.trade_fee_schema(),
+                    trade_type=tracked_order.trade_type,
+                    percent_token=percent_token,
+                    flat_fees=[TokenAmount(amount=Decimal(execution_event["commission"]),
+                                           token=percent_token)]
+                )
+                trade_update = TradeUpdate(
+                    trade_id=execution_event["id"],
+                    client_order_id=tracked_order.client_order_id,
+                    exchange_order_id=order_id,
+                    trading_pair=tracked_order.trading_pair,
+                    fee=fee,
+                    fill_base_amount=Decimal(execution_event["quantity"]),
+                    fill_quote_amount=Decimal(execution_event["quantity"]) * Decimal(execution_event["rate"]),
+                    fill_price=Decimal(execution_event["rate"]),
+                    fill_timestamp=_get_timestamp(execution_event["executedAt"]),
+                )
+                self._order_tracker.process_trade_update(trade_update)
+
+    def _get_fee(self,
+                 base_currency: str,
+                 quote_currency: str,
+                 order_type: OrderType,
+                 order_side: TradeType,
+                 amount: Decimal,
+                 price: Decimal = s_decimal_NaN,
+                 is_maker: Optional[bool] = None) -> TradeFeeBase:
+        is_maker = is_maker or (order_type is OrderType.LIMIT_MAKER)
+        fee = build_trade_fee(
+            self.name,
+            is_maker,
+            base_currency=base_currency,
+            quote_currency=quote_currency,
+            order_type=order_type,
+            order_side=order_side,
+            amount=amount,
+            price=price,
+        )
+        return fee
+
+    def _get_new_client_order_id(self,
+                                 is_buy: bool,
+                                 trading_pair: str,
+                                 hbot_order_id_prefix: str,
+                                 max_id_len: int
+                                 ):
+        return str(uuid.uuid4())
```

Comment:
We should move this function to the connectors' utils module

Reply:
This function is a exchangepybase function which is overriden.